include/mxnet/random_generator.h

```diff
@@ -71,14 +71,23 @@ class RandGenerator<cpu, DType> {
       return dist_uniform(*engine_);
     }
 
+    MSHADOW_XINLINE int discrete_uniform(const int lower, const int upper) {
```

Comment:
Does it only support int32? Does the test case pass if you have randint with range 4B to 4B + 10?

Reply:
updated with int64_t